### PR TITLE
[AWS Fargate] Execution role required if CloudWatch log group is specified

### PR DESCRIPTION
**Pull request by ewbankkit**
Fixes https://github.com/virtual-kubelet/virtual-kubelet/issues/193.
I still need to verify with e2e tests.
~~Shall I add some unit tests for configuration file validation?~~

### CODE REVIEW

File: providers/aws/fargate.toml [hunk 1]
```diff
@@ -32,6 +32,7 @@ ExecutionRoleArn = ""
 
 # Amazon CloudWatch log group name used to store container logs. Optional.
 # If omitted, container logs will not be available.
+# If specified, execution role is required.
```
Comment (ofiliz) [inline providers/aws/fargate.toml]: Minor: We can provide more info here:
"If specified, an execution role with access to CloudWatch logs is required."

Comment (ewbankkit) [reply providers/aws/fargate.toml]: OK, I'll update.

**Comment by ewbankkit** [comment]
Ran e2e tests:

```console
$ VK_TEST_FARGATE_REGION=us-west-2 go test
Starting provider tests in region us-west-2
Tagging: vpc-821e2bfb
Tagging: subnet-2ef09f74
Tagging: igw-f01e6696
Tagging: rtb-e1073699
Fargate provider test configuration:
Region = "us-west-2"
ClusterName = "vk-fargate-e2e-test"
Subnets = [ "subnet-2ef09f74" ]
SecurityGroups = [ ]
AssignPublicIPv4Address = true
ExecutionRoleArn = "vk-fargate-e2e-test"
CloudWatchLogGroupName = "/ecs/vk-fargate-e2e-test"
2018/05/10 09:11:24 Creating Fargate provider.
2018/05/10 09:11:24 Loaded provider configuration file /var/folders/4j/lf5jdq5j4y91g80ytf6qzjwmk6drp5/T/example080440368.
2018/05/10 09:11:24 Created Fargate service client.
2018/05/10 09:11:24 Looking for Fargate cluster vk-fargate-e2e-test in region us-west-2.
2018/05/10 09:11:25 Found Fargate cluster vk-fargate-e2e-test in region us-west-2.
...
ok  	github.com/virtual-kubelet/virtual-kubelet/providers/aws	103.208s
```

**Comment by ewbankkit** [comment]
Removed _WIP_.
Merge when happy.

**Pull request merged**